From 1a3cb2750e4df54abb678ad48bef77cdfad6df5a Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Thu, 7 May 2020 20:41:19 +0530
Subject: [PATCH 35/37] octeontx2-af: Do not change maximum NPC interfaces

commit c0193bebea210e2627a18c95d7943382048243c4 from
git@git.assembla.com:cavium/WindRiver.linux.git

As per the software design for loading profiles from
firmware, changing the maximum interfaces macro causes
incompatibility with existing profile binaries.
Hence the macro is reverted.

Change-Id: I2cdd28ed50f6692e0bad02b734f8c5cdee5a0ad1
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/28052
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/mbox.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/mbox.h b/drivers/net/ethernet/marvell/octeontx2/af/mbox.h
index 4ce0d63de89a..b94777ad2e36 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/mbox.h
+++ b/drivers/net/ethernet/marvell/octeontx2/af/mbox.h
@@ -1309,7 +1309,7 @@ struct npc_get_kex_cfg_rsp {
 	struct mbox_msghdr hdr;
 	u64 rx_keyx_cfg;   /* NPC_AF_INTF(0)_KEX_CFG */
 	u64 tx_keyx_cfg;   /* NPC_AF_INTF(1)_KEX_CFG */
-#define NPC_MAX_INTF	4
+#define NPC_MAX_INTF	2
 #define NPC_MAX_LID	8
 #define NPC_MAX_LT	16
 #define NPC_MAX_LD	2
-- 
2.17.1

